astropy/io/fits/tests/test_connect.py

```diff
@@ -64,6 +66,14 @@ def setup_class(self):
             dtype=[("a", int), ("b", "U1"), ("c", float)],
         )
 
+    def test_overwrite_with_path(self, tmp_path):
+        filename = "temp.fits"
+        t1 = Table(self.data)
+        with contextlib.chdir(tmp_path):
+            t1.write(filename, format="fits")
+            t1.write(Path(filename), format="fits", overwrite=True)
+        t1.write(Path(tmp_path / filename), format="fits", overwrite=True)
```

Comment:
Is this second overwrite still relevant with the context manager above?
```suggestion
```

Reply:
Nvm I just saw Simon's comment above. Ignore this.

Reply:
I was following [Simon's suggestion](https://github.com/astropy/astropy/pull/17552#discussion_r1891343601) ... at least this was my undertanding of his comment... :grin: